cmd/device-worker/main.go

```diff
@@ -150,6 +154,28 @@ func main() {
 	if err := s.Serve(l); err != nil {
```

Comment:
I think that `s.Serve(l)` is blocking and the code after it will only be called when the process is being stopped. Is that the goal?

Reply:
No, I need a blocking code for the channel 
what do you suggest me to do?

Reply:
Would wrapping your code in a goroutine and running it before `s.Serve(l)` work correctly?